src/main/scala/uk/gov/nationalarchives/omega/api/messages/LocalMessageStore.scala

```diff
@@ -56,13 +56,21 @@ class LocalMessageStore(directoryPath: Path) {
     )
 
   def readAllFilesInDirectory(): IO[List[LocalMessage]] =
-    directoryPath.toFile
-      .listFiles()
-      .toList
-      .traverse { path =>
-        deserializeFile(path.toPath).map(_.toOption)
-      }
-      .map(_.flatten)
+    IO.blocking {
+      directoryPath.toFile
+        .listFiles()
+        .toList
+    }.flatMap(
+      _.traverse(path =>
+        deserializeFile(path.toPath).map {
+          case Success(s) => Some(s)
+          case Failure(e) =>
+            logger.info(s"Error during reading files in directory $e")
```

Comment:
Can we include the path in the error message please.

Reply:
Done